src/content/developers/docs/smart-contracts/formal-verification/index.md

```diff
@@ -0,0 +1,136 @@
+
+## Disadvantages of formal verification {#disadvantages-of-formal-verification}
+
+Although formal verification is great for testing smart contracts, it has some disadvantages:
+
+### 1. Difficulty in creating formal specifications {#difficulty-in-creating-formal-specifications}
+
+A prerequisite for formal verification is creating formal specifications, which can take a lot of time and effort. This will likely be the most difficult part of using formal verification for smart contracts.
+
+### 2. False negatives {#false-negatives}
+
+Formal verification can only check if the execution of the smart contract matches formal specifications. If specifications are poorly written, certain vulnerabilities cannot be detected by the formal verification audit. In this case, a developer might erroneously assume that the contract is bug-free.
+
+## Use formal verification tools {#use-formal-verification-tools}
+
+### Creating formal specifications {#creating-formal-specifications}
+
+**Act - _High-level specification language for defining properties for EVM smart contracts._**
+
+- [GitHub](https://github.com/ethereum/act)
+- [Documentation](https://ethereum.github.io/act/)
+
+**Scribble -** **_A specification language tool for translating high-level specifications into Solidity code._**
+
+- [Website](https://consensys.net/diligence/scribble/)
+- [Documentation](https://docs.scribble.codes/language/introduction)
+
+### Fuzzing smart contracts {#fuzzing-smart-contracts}
+
+**Echidna - _A fast contract fuzzer for detecting vulnerabilities in smart contracts through property-based testing._**
+
+- [GitHub](https://github.com/crytic/echidna/)
+
+**Harvey - _Automated fuzzing tool useful for detecting property violations in smart contract code._**
+
+- [Website](https://consensys.net/diligence/fuzzing/)
+
+### Code analysis {#code-analysis}
+
+**Oyente - _Static analysis tool for analyzing vulnerabilities in EVM bytecode with symbolic execution._**
+
+- [GitHub](https://github.com/melonproject/oyente)
+
+**Mythril - _EVM bytecode assessment tool for detecting contract vulnerabilities using taint analysis, concolic analysis, and control flow checking._**
+
+- [GitHub](https://github.com/ConsenSys/mythril-classic)
+- [Documentation](https://mythril-classic.readthedocs.io/en/master/about.html)
+
+**Manticore _- A tool for analyzing EVM bytecode analysis tool based on symbolic execution._**
+
+- [GitHub](https://github.com/trailofbits/manticore)
+- [Documentation](https://github.com/trailofbits/manticore/wiki)
+
+**Securify** **_- A security analyzer for analyzing Solidity source code in Ethereum smart contracts and detecting property violations._**
+- [GitHub](https://github.com/eth-sri/securify2)
+
+### Formal verification projects in Ethereum {#formal-verification-projects-in-ethereum}
```

Comment:
What is the exact meaning of this section? The tools in the previous section also fit under the "eth fv" umbrella.

Reply:
I was aiming to highlight tools that provide a more comprehensive framework for formal verification. For example, the KEVM framework was the most visible project I found that was developed specifically for formal verification. But if you want, I could have them listed together under a "Formal verification tools for Ethereum smart contracts" heading.